### PR TITLE
Add `VerificationMessage` impl to `TrustedIdentitiesVerifier`

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
The `TrustedIdentitiesVerifier` can now be displayed as part of a
`VerificationTree`

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #125** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/125" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/125?utm_source=stack-comment).

File: verifier/src/identity.rs [hunk 1]
```diff
@@ -299,96 +328,212 @@ where
...
+        f: &mut Formatter<'_>,
+        pad: usize,
+        identity: &IdentityOutput,
+    ) -> core::fmt::Result {
+        // We re-verify here, in order to leverage the `fmt_padded` method on the `Verifier` trait
+        match self {
+            Self::MrEnclave(verifier) => {
+                let result = verifier.verify(identity);
+                verifier.fmt_padded(f, pad, &result)
+            }
```
Comment (nick-mobilecoin) [inline verifier/src/identity.rs:347]: I couldn't think of another way to get the output without re-verifying. I don't think we'll have that many identities and the overall comparisons should be pretty cheap.  And this will only be done when displaying the results.

File: verifier/src/identity.rs [hunk 2]
```diff
@@ -299,96 +328,212 @@ where
 /// identity, while the `TrustedIdentitiesVerifier` is for verifying one of *multiple* identities.
 #[derive(Debug, Clone)]
 enum TrustedIdentityVerifier {
-    MrEnclave(TrustedMrEnclaveIdentityVerifier),
-    MrSigner(TrustedMrSignerIdentityVerifier),
+    MrEnclave(And<MrEnclaveVerifier, AdvisoriesVerifier>),
+    MrSigner(And<MrSignerVerifier, AdvisoriesVerifier>),
```
Comment (nick-mobilecoin) [inline verifier/src/identity.rs:332]: I dropped the dedicated structs and used the And types directly here, mainly to thin out the types, there are quite a few in this file and it's getting hard to follow.

File: verifier/src/identity.rs [hunk 3]
```diff
@@ -609,5 +771,49 @@ mod test {
...
+              - MRENCLAVE: 0101010101010101010101010101010101010101010101010101010101010101
+              - MRSIGNER key hash: 0202020202020202020202020202020202020202020202020202020202020202
+              - ISV product ID: 3
+              - ISV SVN: 4
+              - advisories: IDs: {"four", "one", "three", "two"} Status: ConfigurationAndSWHardeningNeeded
+              Searched through the following identities:
+              - [ ] Both of the following must be true:
+                - [ ] The MRENCLAVE should be 0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b, but the actual MRENCLAVE was 0101010101010101010101010101010101010101010101010101010101010101
+                - [x] The allowed advisories are IDs: {"four", "one", "three", "two"} Status: ConfigurationAndSWHardeningNeeded
+              - [ ] Both of the following must be true:
```
Comment (nick-mobilecoin) [inline verifier/src/identity.rs:787]: Up for suggestions on better ways to separate out the individual identity attempts, at the same time this is probably good enough.

File: verifier/src/identity.rs [hunk 4]
```diff
@@ -268,28 +271,54 @@ where
...
+        }
+
+        let status = crate::choice_to_status_message(is_success);
+        writeln!(f, "{:pad$}{status} No enclave identity matched for:", "")?;
+
+        let pad = pad + MESSAGE_INDENT;
+        result.value.fmt_padded(f, pad)?;
+
+        let TrustedIdentityValue::Identity(identity) = &result.value else {
+            panic!("Should have an IdentityOutput if we failed to verify");
```
Comment (eranrund) [inline verifier/src/identity.rs:312]: Should or should not?

Comment (eranrund) [reply verifier/src/identity.rs:312]: Oops, I misread it and missed the `else.

**Review by eranrund — APPROVED** [review]
